spec/enumerable_spec.rb

```diff
@@ -0,0 +1,289 @@
+# spec/enumerable_spec.rb
+
+require '../enumerable.rb'
+
+describe Enumerable do
+  let(:num_array) { [1, 2, 3, 5, 7, 8] }
+  let(:empty_array) { [] }
+  let(:str_array) { %w[c h o c o m i n t] }
+  let(:range) { (1..10) }
+  let(:hash) { { 'a': 2, 'b': 5, 'c': 7, 'd': 8 } }
+  let(:str) { %(pineapple) }
+  let(:block_for_num) { proc { |i| i * 2 } }
+  let(:block_for_str) { proc { |item| item } }
+  let(:block_for_hash) { proc { |key, value| "#{key} is #{value}" } }
+  let(:bk_for_each_idx) { proc { |item, index| "#{index}: #{item}" } }
+  let(:bk_condit_num) { proc { |i| i.even? } }
+  let(:bk_condit_str) { proc { |item| item.is_a?(String) } }
+  let(:bk_condit_hash) { proc { |_key, value| value < 3 } }
+  let(:block_inject_num) { proc { |accum, i| accum * i } }
+  let(:block_inject_str) { proc { |accum, word| accum.size > word.size ? accum : word } }
+
+  describe '#my_each' do
+    it 'returns the same result as each method to num_array' do
+      expect(num_array.my_each(&block_for_num)).to eql(num_array.each(&block_for_num))
+    end
```

Comment:
`#each` **returns** the enumerable it's called on. In this test, you are basically checking if `#my_each` returns